apps/snitch_core/lib/core/data/schema/user.ex

```diff
@@ -50,31 +50,55 @@ defmodule Snitch.Data.Schema.User do
     timestamps()
   end
 
-  @required_fields ~w(first_name last_name email is_admin?)a
-  @password_fields ~w(password password_confirmation)a
+  @create_fields ~w(first_name last_name email password password_confirmation is_admin)a
+  @update_fields ~w(sign_in_count failed_attempts)a ++ @create_fields
+
 
-  @spec registration_changeset(__MODULE__.t(), map) :: Ecto.Changeset.t()
   @doc """
-  Returns a changeset to register a new user
+  Returns a complete changeset with totals.
+
+  The `action` field can be either `:create` or `:update`.
+
+  * `:create`
+    - A map with fields first_name, last_name, email, password,
+      and password_confirmation is expected.
+  * `:update`
+    - No required fields.
+
+  ## Note
+  The changeset `action` is not set.
   """
-  def registration_changeset(user, params) do
+
+  @spec changeset(__MODULE__.t(), map, :create | :update) :: Ecto.Changeset.t()
+  def changeset(user, params, action) do
     user
-    |> changeset(params)
-    |> validate_required(@password_fields)
+    |> cast(params, @create_fields ++ @update_fields)
     |> validate_confirmation(:password)
     |> validate_password(:password)
-    |> put_pass_hash()
+    |> validate_format(:email, ~r/@/)
+    |> do_changeset(action)
   end
 
-  @spec changeset(__MODULE__.t(), map) :: Ecto.Changeset.t()
-  def changeset(user, params) do
-    user
-    |> cast(params, @required_fields)
-    |> validate_required(@required_fields)
-    |> validate_format(:email, ~r/@/)
-    |> unique_constraint(:email)
+  @spec create_changeset(Ecto.Changeset.t()) :: Ecto.Changeset.t()
+  defp create_changeset(user_changeset) do
+    user_changeset
+    |> validate_required(@create_fields)
+    |> put_pass_hash
+  end
+
+  @spec update_changeset(Ecto.Changeset.t()) :: Ecto.Changeset.t()
+  defp update_changeset(%{changes: changes} = user_changeset) do
+    cond do
+      Map.has_key?(changes, :password) ->
+        user_changeset |> put_pass_hash
+
+      true -> user_changeset
+    end
   end
 
+  defp do_changeset(changeset, :create), do: create_changeset(changeset)
+  defp do_changeset(changeset, :update), do: update_changeset(changeset)
+
   defp validate_password(changeset, field) do
```

Comment:
Could you please replace this function with a simple `validate_length(:password, min: @password_min_length)`?